nextcord/polls.py

```diff
@@ -0,0 +1,440 @@
+            "answers": [answer.to_dict() for answer in self.answers],
+            "duration": self.duration,
+            "allow_multiselect": self.allow_multiselect,
+        }
+        if self.layout_type:
+            payload["layout_type"] = self.layout_type.value
+        return payload
+
+    def add_answer(self, text: str, emoji: Optional[EmojiInputType] = None) -> Self:
+        """Add a choice i.e. answer to the poll.
+
+        This function returns the class instance to allow for fluent-style chaining.
+
+        Parameters
+        ----------
+        text: :class:`str`
+            The text description of the choice.
+        emoji: Optional[Union[:class:`PartialEmoji`, :class:`Emoji`, :class:`str`]]
+            The emoji description of the choice.
+        """
+        self.answers.append(PollAnswer(poll_media=PollMedia(text=text, emoji=emoji)))
+        return self
+
+
+class Poll:
+    """A poll object.
+
+    .. versionadded:: 3.0
+
+    Attributes
+    ----------
+    message: :class:`Message`
+        The message this poll belong to.
+    question: :class:`PollMedia`
+        The question being asked in a poll.
+    answers: List[:class:`PollAnswer`]
+        The answers of the poll.
+    expiry: :class:`datetime.datetime`
+        The expiry of the poll.
+    allow_multiselect: :class:`bool`
+        Whether this poll allow people to select more than 1 option.
+    layout_type: :class:`PollLayoutType`
+        The layout type of the poll.
+    results: Optional[:class:`PollResults`]
+        The results of the poll.
+    """
+
+    __slots__ = (
+        "message",
+        "question",
+        "_state",
+        "answers",
+        "expiry",
+        "allow_multiselect",
+        "layout_type",
+        "results",
+    )
+
+    def __init__(self, data: PollData, *, message: Message, state: ConnectionState) -> None:
+        self.message: Message = message
+        self._state: ConnectionState = state
+        self.question: PollMedia = PollMedia.from_dict(data["question"])
+        self.answers: List[PollAnswer] = [
+            PollAnswer.from_dict(poll_answer) for poll_answer in data["answers"]
+        ]
+        self.expiry: datetime = datetime.fromisoformat(data["expiry"])
+        self.allow_multiselect: bool = data["allow_multiselect"]
+        self.layout_type: PollLayoutType = try_enum(PollLayoutType, data["layout_type"])
+
+        self.results: Optional[PollResults] = (
+            PollResults(data["results"], self) if "results" in data else None
+        )
+
+    def __repr__(self) -> str:
+        return (
+            "<Poll "
+            f"message={self.message!r} "
+            f"question={self.question!r} "
+            f"answers={self.answers!r} "
+            f"expired={self.expired!r}"
+            ">"
+        )
+
+    def __eq__(self, other) -> bool:
+        return isinstance(other, self.__class__) and other.message.id == self.message.id
+
+    async def expire(self) -> Message:
```

Comment:
I'm not entirely happy with this method name. It isn't expiring, you're manually ending it. Given that the endpoint is named "End Poll", I think `end` would be a better name for this?